scripts/core/api/api-service.js

```diff
@@ -52,13 +54,41 @@ function APIProvider() {
             return response.status >= 200 && response.status < 300 && !isErrData(response.data);
         }
 
+        var cache = {};
+
         /**
          * Call $http once url is resolved
+         *
+         * Detect duplicate requests and serve these from cache.
          */
         function http(config) {
             return $q.when(config.url).then(function(url) {
                 config.url = url;
-                return $http(config);
+
+                if (config.method !== 'GET') {
+                    return $http(config);
+                }
+
+                let now = Date.now();
+                let key = config.url + angular.toJson(config.params || {});
+                let last = cache[key] || null;
+                if (last && now - last.now < 100) {
```

Comment:
In my opinion `100` should be a const at the top, so it's clear and easy to config. It would also give the value meaning.